test/ruby/libdnf/support/support.rb

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2020-2021 Red Hat, Inc.
+#
+# This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+#
+# Libdnf is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Libdnf is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+
+
+require 'test/unit'
+require 'tmpdir'
+include Test::Unit::Assertions
+
+require 'libdnf/base'
+require 'libdnf/rpm'
+
+
+PROJECT_BINARY_DIR = ENV["PROJECT_BINARY_DIR"]
+PROJECT_SOURCE_DIR = ENV["PROJECT_SOURCE_DIR"]
+
+
+class LibdnfTestCase < Test::Unit::TestCase
+
+    def setup()
+        @base = Base::Base.new()
+
+        @cachedir = Dir.mktmpdir("libdnf-test-ruby-")
+        @base.get_config().cachedir().set(Conf::Option::Priority_RUNTIME, @cachedir)
+
+        @repo_sack = @base.get_repo_sack()
+        @package_sack = @base.get_rpm_package_sack()
+    end
+
+    def teardown()
+        FileUtils.remove_entry(@cachedir)
+    end
+
+    def _add_repo(repoid, repo_path)
+        repo = @repo_sack.new_repo(repoid)
+        repo.get_config().baseurl().set(Conf::Option::Priority_RUNTIME, "file://" + repo_path)
+        repo.load()
+        @package_sack.load_repo(repo.get(),
+            Rpm::PackageSack::LoadRepoFlags_USE_FILELISTS |
+            Rpm::PackageSack::LoadRepoFlags_USE_OTHER |
+            Rpm::PackageSack::LoadRepoFlags_USE_PRESTO |
+            Rpm::PackageSack::LoadRepoFlags_USE_UPDATEINFO
+        )
+    end
+
+    def add_repo_rpm(repoid)
```

Comment:
Please can you add comments like I requested for the python code?

Reply:
Done